File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def get_power(num):
    cnt = 0
    for i in range(1, int(num ** (0.5)) + 1):
        if num % i == 0:
            cnt += 1
            if (i ** 2 != num) :
                cnt += 1
    return cnt

def solution(number, limit, power):
    answer = 0
    for i in range(1, number + 1):
        k = get_power(i)
        answer += k if k <= limit else power
    return answer

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <numeric>

using namespace std;

int get_power(int num) {
    int cnt = 0;
    for (int i = 1; i < (int)(pow(num, 0.5)) + 1; i++) {
        if (num % i == 0) {
            cnt++;
            if ((int)pow(i, 2) != num)
                cnt++;
        }
    }
    return cnt;
}

int solution(int number, int limit, int power) {
    int answer = 0;
    for (int i = 1; i <= number; i++) {
        int p = get_power(i);
        (p > limit) ? answer += power : answer += p;
    }
    return answer;
}